[ML][Fleet] Adds link to anomaly detection configurations from Integration > Assets tab

alvarezmelissa87 opened the pull request:
## Summary

Related issue: https://github.com/elastic/kibana/issues/182199?reload=1

This PR adds a link to `ML > Anomaly Detection > Supplied Configurations` from `Integration > Assets tab` for 'ML Modules' assets. 

The naming of the asset is also updated in Fleet to be consistent with the ML UI.

<img width="429" alt="Screenshot 2024-09-19 at 13 46 47" src="https://github.com/user-attachments/assets/9fcc3606-cc08-483f-88b4-00c07de3fc57">

<img width="717" alt="Screenshot 2024-09-19 at 13 47 52" src="https://github.com/user-attachments/assets/847f3d6e-95a1-491c-aa3e-7d54da7df98a">


### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)




<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

peteharverson commented on x-pack/plugins/fleet/public/applications/integrations/sections/epm/constants.tsx:
```diff
@@ -66,10 +66,10 @@ export const AssetTitleMap: Record<
   ),
   'ml-module': i18n.translate('xpack.fleet.epm.assetTitles.mlModules', {
-    defaultMessage: 'ML modules',
+    defaultMessage: 'Anomaly Detection supplied configurations',
```
I think we can drop 'supplied' from this text to the more concise `Anomaly detection configurations`.

alvarezmelissa87 replied:
updated in 3c72ce197202467effe812c84029eb2b3ca89b53

peteharverson commented on x-pack/plugins/fleet/public/applications/integrations/sections/epm/constants.tsx:
```diff
@@ -66,10 +66,10 @@ export const AssetTitleMap: Record<
   }),
   ml_module: i18n.translate('xpack.fleet.epm.assetTitles.mlModules', {
-    defaultMessage: 'ML modules',
+    defaultMessage: 'Anomaly Detection supplied configurations',
```
As above, suggest we go with `Anomaly detection configurations` here.

alvarezmelissa87 replied:
Updated in 3c72ce197202467effe812c84029eb2b3ca89b53

szabosteve commented on x-pack/plugins/fleet/public/applications/integrations/sections/epm/constants.tsx:
```diff
@@ -66,10 +66,10 @@ export const AssetTitleMap: Record<
   }),
   ml_module: i18n.translate('xpack.fleet.epm.assetTitles.mlModules', {
-    defaultMessage: 'ML modules',
+    defaultMessage: 'Anomaly Detection configurations',
```
Nit. As above.

```suggestion
    defaultMessage: 'Anomaly detection configurations',
```

alvarezmelissa87 replied:
Updated in f4cde98229ffc604f8b039d8595c396a70637b20

alvarezmelissa87 commented:
This has been updated and is ready for a final look when you get a chance 🙏 cc @peteharverson, @szabosteve

peteharverson reviewed (approved):
Latest changes LGTM

alvarezmelissa87 commented:
@elasticmachine merge upstream

walterra commented on x-pack/plugins/ml/public/application/supplied_configurations/supplied_configurations.tsx:
```diff
@@ -82,22 +101,46 @@ export const SuppliedConfigurations = () => {
-    fields: {
-      tags: {
-        type: 'string',
+  const schema = useMemo(
+    () => ({
+      strict: true,
+      fields: {
+        tags: {
+          type: 'string',
+        },
       },
+    }),
+    []
+  );
```
This looks like it's just initialized once without any options, could it be moved outside the component without the `useMemo()`?

alvarezmelissa87 replied:
Oh good catch - updated in 643488707a8c517acfb689574f11f4024fc53c9b

walterra reviewed (approved):
LGTM, just added a question about `schema`. About `stringMatch`: Would be good to add some unit tests that work with the update but would fail with the previous version. In a follow we could also consolidate the duplicate code into a package.

alvarezmelissa87 commented:
> LGTM, just added a question about `schema`. About `stringMatch`: Would be good to add some unit tests that work with the update but would fail with the previous version. In a follow we could also consolidate the duplicate code into a package.

Added unit tests in 643488707a8c517acfb689574f11f4024fc53c9b
Agree on moving to a package in follow-up work 👍

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/193485">](https://github.com/elastic/kibana/pull/193485)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
